deepchem/models/layers.py

```diff
@@ -370,6 +370,471 @@ def call(self, inputs):
+               activation: Callable = activations.tanh,
+               dropout_rate: float = 0.0,
+               edges: int = 5,
+               name: str = "",
+               **kwargs):
+    """
+    Initialize the layer
+
+    Parameters
+    ---------
+    units: Tuple, optional (default=(128,64)), min_length=2
+      List of dimensions used by consecutive convolution layers.
+      The more values the more convolution layers invoked.
+    activation: function, optional (default=tanh)
+      activation function used across model, default is Tanh
+    dropout_rate: float, optional (default=0.0)
+      Used by dropout layer
+    edges: int, optional (default=0)
+      Controls how many dense layers use for single convolution unit.
+      Typically matches number of bond types used in the molecule.
+    name: string, optional (default="")
+      Name of the layer
+    """
+
+    super(MultiGraphConvolutionLayer, self).__init__(name=name, **kwargs)
+    assert len(units) > 1, "Layer requires at least two values"
+
+    self.units = units
+    self.activation = activation
+    self.dropout_rate = dropout_rate
+    self.edges = edges
+
+    self.first_convolution = GraphConvolutionLayer(
+        self.units[0], self.activation, self.dropout_rate, self.edges)
+    self.gcl = [
+        GraphConvolutionLayer(u, self.activation, self.dropout_rate, self.edges)
+        for u in self.units[1:]
+    ]
+
+  def call(self, inputs, training=False):
+    """
+    Invoke this layer
+
+    Parameters
+    ----------
+    inputs: list
+      List of two input matrices, adjacency tensor and node features tensors
+      in one-hot encoding format.
+    training: bool
+      Should this layer be run in training mode.
+      Typically decided by main model, influences things like dropout.
+
+    Returns
+    --------
+    convolution tensor: tf.Tensor
+      Result of input tensors going through convolution a number of times.
+    """
+
+    adjacency_tensor = inputs[0]
+    node_tensor = inputs[1]
+
+    tensors = self.first_convolution([adjacency_tensor, node_tensor])
+
+    for layer in self.gcl:
+      tensors = layer(tensors)
+
+    _, _, hidden_tensor = tensors
+
+    return hidden_tensor
+
+  def get_config(self) -> Dict:
+    """
+    Returns config dictionary for this layer.
+    """
+
+    config = super(MultiGraphConvolutionLayer, self).get_config()
+    config["units"] = self.units
+    config["activation"] = self.activation
+    config["dropout_rate"] = self.dropout_rate
+    config["edges"] = self.edges
+    return config
+
+
+class GraphEncoderLayer(tf.keras.layers.Layer):
+  """
+  Main learning layer used by MolGAN model.
+  MolGAN is a WGAN type model for generation of small molecules.
+  It role is to further simplify model.
+  This layer can be manually built by stacking graph convolution layers
+  followed by graph aggregation.
+
+  Example
+  --------
+  def create_discriminator(adjacency_tensor, node_tensor):
```

Comment:
To have this render correctly you'd want to do this as
```
Example
------
>>> def create_discriminator(...)
...
```
That is, with the `>>>` added before the code sample

Reply:
OK, I have added per your suggestion and indented the code to match.
Should be fine now. 
I have been thinking and I will probably modify layers slightly once everything is in place (layers, model etc.)
I have been thinking about improving overall flow of the code.
This is something I will pick your brain about once we release the code.